File: Lecture05/ListViews/app/src/main/java/com/codingblocks/listviews/adapters/CourseAdapter.java
package com.codingblocks.listviews.adapters;

import android.content.Context;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.TextView;

import com.codingblocks.listviews.R;
import com.codingblocks.listviews.models.Course;

import java.util.ArrayList;

/**
 * Created by arnav on 12/23/2017.
 */

public class CourseAdapter extends BaseAdapter{
    public static final String TAG = "LIST";

    private ArrayList<Course> courses;
    private Context context;

    public CourseAdapter(Context context, ArrayList<Course> courses) {
        this.context = context;
        this.courses = courses;
    }

    @Override
    public int getCount() {
        return courses.size();
    }

    @Override
    public Course getItem(int position) {
        return courses.get(position);
    }

    @Override
    public long getItemId(int position) {
        return 0;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        Log.d(TAG, "position = " + position + "convertView: = " + convertView);
        if (convertView == null) {
            LayoutInflater li = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
            convertView = li.inflate(R.layout.list_item_course, parent, false);

        }
        Course currentCourse = getItem(position);
        ((TextView) convertView.findViewById(R.id.tvCourseName)).setText(currentCourse.getName());
        ((TextView) convertView.findViewById(R.id.tvTeacherName)).setText(currentCourse.getTeacherName());
        ((TextView) convertView.findViewById(R.id.tvLectures)).setText(currentCourse.getLecturesAsString());


        return convertView;
    }
}

File: Lecture06/KotlinSample/app/src/main/java/com/codingblocks/kotlinsample/adapters/CourseAdapter.kt
package com.codingblocks.kotlinsample.adapters

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import com.codingblocks.kotlinsample.R
import com.codingblocks.kotlinsample.models.Course
import kotlinx.android.synthetic.main.list_item_course.view.*

/**
 * Created by arnav on 12/24/2017.
 */
class CourseAdapter(val context: Context, val courses: ArrayList<Course>): RecyclerView.Adapter<CourseAdapter.CourseViewHolder>() {


    override fun onCreateViewHolder(parent: ViewGroup?, viewType: Int): CourseViewHolder {
        val li: LayoutInflater = context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
        val itemView = li.inflate(R.layout.list_item_course, parent, false)
        return CourseViewHolder(itemView)
    }

    override fun onBindViewHolder(holder: CourseViewHolder?, position: Int) {
        holder?.bindView(courses[position])
    }

    override fun getItemCount(): Int {
        return courses.size
    }

    inner class CourseViewHolder(itemView: View?) : RecyclerView.ViewHolder(itemView) {

        fun bindView (course: Course) {
            itemView.tvCourseName.text = course.name
            itemView.tvTeacherName.text = course.teacherName
            itemView.tvLectures.text = course.lectures.toString()
        }


    }
}